src/System.Text.Primitives/System/Text/Encoding/SymbolTable.Utf8.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Text
+{
+    public partial class SymbolTable
+    {
+        private sealed class Utf8InvariantSymbolTable : SymbolTable
+        {
+            private static readonly byte[][] Utf8DigitsAndSymbols = new byte[][]
+            {
+                new byte[] { 48, },
+                new byte[] { 49, },
+                new byte[] { 50, },
+                new byte[] { 51, },
+                new byte[] { 52, },
+                new byte[] { 53, },
+                new byte[] { 54, },
+                new byte[] { 55, },
+                new byte[] { 56, },
+                new byte[] { 57, }, // digit 9
+                new byte[] { 46, }, // decimal separator
+                new byte[] { 44, }, // group separator
+                new byte[] { 73, 110, 102, 105, 110, 105, 116, 121, },
+                new byte[] { 45, }, // minus sign
+                new byte[] { 43, }, // plus sign
+                new byte[] { 78, 97, 78, }, // NaN
+                new byte[] { 69, }, // E
+                new byte[] { 101, }, // e
+            };
+
+            public Utf8InvariantSymbolTable() : base(Utf8DigitsAndSymbols) {}
+
+            public override bool TryEncode(byte utf8, Span<byte> destination, out int bytesWritten)
+            {
+                if (destination.Length < 1)
```

Comment:
I wonder if we would not get better codegen if:
1. First, we did a check for the utf8 range.
2. Then we did a check for length, and immediately after wrote to the span.

The reason is that span does the exact same length check, which might be easier to eliminate if the checks endup being one after the other.